### PR TITLE
feat: add optional response images with cloudinary

### PR DESCRIPTION
**Pull request by artis3n**
I was trying to adapt this to this theme and didn't see another option already available, so I thought you might like this PR. Let me know if this isn't the type of change you want to include in the theme.

This optionally provides a capability for responsive images via Cloudinary. It requires images to be hosted in Cloudinary (I have a free account for my blog). It does not add any weight if someone does not wish to use the feature. If they would like to use this, uncommenting `cloudinary_cloud_name` will load in the necessary javascript.

I am using it like this on my markdown articles:
image

Notably, the `src` path is the remote path in Cloudinary where my image is located, not a local path. This cloudinary configuration will also auto-select the best image type for the user's browser - e.g. AVIF or webp if supported, else png, jpg, etc. So a file extension is excluded from the path in my example.

The new `meta` tag addition to `head.html` is to gather mobile device dimensions so images can be properly sized in those cases.

### CODE REVIEW

File: layouts/shortcodes/dynamic-img.html [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+    {{< dynamic-img src="/my/image/on/cloudinary.webp" title="A title for the image" >}}
+
+    Optionally, you can customize the width or other general styles for the image:
+    {{< dynamic-img src="/my/image/on/cloudinary.webp" title="A title for the image" width="w_auto" style="max-width:90%" >}}
+-->
+{{ $image := .Get "src" }}
+{{ $alt := .Get "title" }}
+{{ $width := .Get "width" | default "w_auto" }}
+{{ $style := .Get "style" | default "max-width:80%" }}
+<img alt="{{ $alt }}" title="{{ $alt }}" data-src="https://res.cloudinary.com/{{ $.Site.Params.cloudinary_cloud_name }}/{{ $width }},c_scale,f_auto,q_auto,dpr_auto{{ $image}}" class="cld-responsive" style="padding-bottom: 16px; {{ $style }}">
```
Comment (gurusabarish) [inline layouts/shortcodes/dynamic-img.html:12]: > 11:14:36 PM: Error: add site dependencies: load resources: loading templates: "/opt/build/repo/layouts/shortcodes/dynamic-img.html:3:1": parse failed: template: shortcodes/dynamic-img.html:3: unexpected "<" in command

please, look into it

Comment (artis3n) [reply layouts/shortcodes/dynamic-img.html:12]: Oh thanks, could you attach the logs to a gist or something? When I browse to that link, I just get this:
image

I'll take a look at why that error is appearing

Comment (gurusabarish) [reply layouts/shortcodes/dynamic-img.html:12]: Here is the [deployment log](https://user-images.githubusercontent.com/53465361/205525316-af878d5c-33e2-4321-966c-0456a3733b94.png)
```
11:14:17 PM: Build ready to start
11:14:25 PM: build-image version: 4c0c1cadee6a31c9bb8d824514030009c4c05c6a (focal)
11:14:25 PM: build-image tag: v4.15.0
11:14:25 PM: buildbot version: 9ef4dba9386902658d5cc7ad8ca0406443896076
11:14:25 PM: Fetching cached dependencies
11:14:25 PM: Starting to download cache of 149.2MB
11:14:27 PM: Finished downloading cache in 1.467376082s
11:14:27 PM: Starting to extract cache
11:14:28 PM: Finished extracting cache in 727.956537ms
11:14:28 PM: Finished fetching cache in 2.241469622s
11:14:28 PM: Starting to prepare the repo for build
11:14:28 PM: Preparing Git Reference pull/80/head
11:14:28 PM: Parsing package.json dependencies
11:14:29 PM: Different publish path detected, going to use the one specified in the Netlify configuration file: 'exampleSite/public' versus 'public' in the Netlify UI
11:14:29 PM: Different build command detected, going to use the one specified in the Netlify configuration file: 'cd exampleSite && hugo --gc --minify --buildFuture --themesDir ../.. -b $DEPLOY_PRIME_URL' versus 'hugo --gc --minify' in the Netlify UI
11:14:29 PM: Section completed: initializing
11:14:29 PM: Starting build script
11:14:29 PM: Installing dependencies
11:14:29 PM: Python version set to 2.7
11:14:29 PM: Started restoring cached node version
11:14:30 PM: Finished restoring cached node version
11:14:30 PM: v12.18.0 is already installed.
11:14:30 PM: Now using node v12.18.0 (npm v6.14.4)
11:14:30 PM: Started restoring cached build plugins
11:14:30 PM: Finished restoring cached build plugins
11:14:30 PM: Attempting ruby version 2.7.1, read from environment
11:14:31 PM: Started restoring cached ruby version
11:14:31 PM: Finished restoring cached ruby version
11:14:31 PM: Using ruby version 2.7.1
11:14:31 PM: Using PHP version 8.0
11:14:31 PM: Installing Hugo 0.92.0
11:14:31 PM: hugo v0.92.0-B3549403+extended linux/amd64 BuildDate=2022-01-12T08:23:18Z VendorInfo=gohugoio
11:14:31 PM: Started restoring cached go cache
11:14:31 PM: Finished restoring cached go cache
11:14:31 PM: Installing Go version 1.14.4 (requested 1.14.4)
11:14:35 PM: unset GOOS;
11:14:35 PM: unset GOARCH;
11:14:35 PM: export GOROOT='/opt/buildhome/.gimme/versions/go1.14.4.linux.amd64';
11:14:35 PM: export PATH="/opt/buildhome/.gimme/versions/go1.14.4.linux.amd64/bin:${PATH}";
11:14:35 PM: go version >&2;
11:14:35 PM: export GIMME_ENV="/opt/buildhome/.gimme/env/go1.14.4.linux.amd64.env"
11:14:35 PM: go version go1.14.4 linux/amd64
11:14:35 PM: Detected 0 framework(s)
11:14:35 PM: Installing missing commands
11:14:35 PM: Verify run directory
11:14:36 PM: ​
11:14:36 PM: ────────────────────────────────────────────────────────────────
11:14:36 PM:   Netlify Build                                                 
11:14:36 PM: ────────────────────────────────────────────────────────────────
11:14:36 PM: ​
11:14:36 PM: ❯ Version
11:14:36 PM:   @netlify/build 28.4.4
11:14:36 PM: ​
11:14:36 PM: ❯ Flags
11:14:36 PM:   baseRelDir: true
11:14:36 PM:   buildId: 638b8af079a0340009297a1b
11:14:36 PM:   deployId: 638b8af079a0340009297a1d
11:14:36 PM: ​
11:14:36 PM: ❯ Current directory
11:14:36 PM:   /opt/build/repo
11:14:36 PM: ​
11:14:37 PM: Creating deploy upload records
11:14:36 PM: ❯ Config file
11:14:36 PM:   /opt/build/repo/netlify.toml
11:14:36 PM: ​
11:14:36 PM: ❯ Context
11:14:36 PM:   deploy-preview
11:14:37 PM: Failed during stage 'building site': Build script returned non-zero exit code: 2 (https://ntl.fyi/exit-code-2)
11:14:36 PM: ​
11:14:36 PM: ────────────────────────────────────────────────────────────────
11:14:36 PM:   1. build.command from netlify.toml                            
11:14:36 PM: ────────────────────────────────────────────────────────────────
11:14:36 PM: ​
11:14:36 PM: $ cd exampleSite && hugo --gc --minify --buildFuture --themesDir ../.. -b $DEPLOY_PRIME_URL
11:14:36 PM: Error: add site dependencies: load resources: loading templates: "/opt/build/repo/layouts/shortcodes/dynamic-img.html:3:1": parse failed: template: shortcodes/dynamic-img.html:3: unexpected "<" in command
11:14:36 PM: Total in 10 ms
11:14:36 PM: ​
11:14:36 PM: ────────────────────────────────────────────────────────────────
11:14:36 PM:   "build.command" failed                                        
11:14:36 PM: ────────────────────────────────────────────────────────────────
11:14:36 PM: ​
11:14:36 PM:   Error message
11:14:36 PM:   Command failed with exit code 255: cd exampleSite && hugo --gc --minify --buildFuture --themesDir ../.. -b $DEPLOY_PRIME_URL (https://ntl.fyi/exit-code-255)
11:14:36 PM: ​
11:14:36 PM:   Error location
11:14:36 PM:   In build.command from netlify.toml:
11:14:36 PM:   cd exampleSite && hugo --gc --minify --buildFuture --themesDir ../.. -b $DEPLOY_PRIME_URL
11:14:36 PM: ​
11:14:36 PM:   Resolved config
11:14:36 PM:   build:
11:14:36 PM:     command: cd exampleSite && hugo --gc --minify --buildFuture --themesDir ../.. -b $DEPLOY_PRIME_URL
11:14:36 PM:     commandOrigin: config
11:14:36 PM:     environment:
11:14:36 PM:       - REVIEW_ID
11:14:36 PM:       - HUGO_VERSION
11:14:36 PM:       - HUGO_THEME
11:14:36 PM:     publish: /opt/build/repo/exampleSite/public
11:14:36 PM:     publishOrigin: config
11:14:37 PM: Caching artifacts
11:14:37 PM: Started saving build plugins
11:14:37 PM: Finished saving build plugins
11:14:37 PM: Started saving pip cache
11:14:37 PM: Finished saving pip cache
11:14:37 PM: Started saving emacs cask dependencies
11:14:37 PM: Finished saving emacs cask dependencies
11:14:37 PM: Started saving maven dependencies
11:14:37 PM: Finished saving maven dependencies
11:14:37 PM: Started saving boot dependencies
11:14:37 PM: Finished saving boot dependencies
11:14:37 PM: Started saving rust rustup cache
11:14:37 PM: Finished saving rust rustup cache
11:14:37 PM: Started saving go dependencies
11:14:37 PM: Finished saving go dependencies
11:14:37 PM: Build failed due to a user error: Build script returned non-zero exit code: 2
11:14:37 PM: Failing build: Failed to build site
11:14:37 PM: Finished processing build request in 12.163983393s
```

Comment (artis3n) [reply layouts/shortcodes/dynamic-img.html:12]: Ah it looks like Hugo is trying to render the example use comment I had. I'll extract that out into documentation elsewhere.

File: .gitignore [hunk 2]
```diff
@@ -16,4 +16,4 @@ hugo.darwin
 hugo.linux
 
 # Temporary lock file while building
-/.hugo_build.lock
```
Comment (artis3n) [inline .gitignore:19]: `cd exampleSite && hugo server --gc --minify --themesDir ../..` resulted in this file showing up under that directory. This change excludes this lock file anywhere it appears.

**Comment by gurusabarish** [comment]
Thanks for the pull request @artis3n ❤️

**Comment by artis3n** [comment]
Thanks! Working through customizations I'm making on my site that require updates to the head or scripts template, so getting them into the theme where I can :)

**Pull request merged**